llama-index-core/llama_index/core/agent/runner/base.py

```diff
@@ -701,6 +701,8 @@ def stream_chat(
                 and chat_response.is_dummy_stream
             )
             e.on_end(payload={EventPayload.RESPONSE: chat_response})
+
+        assert isinstance(chat_response, StreamingAgentChatResponse)
```

Comment:
Why is there a need to check the type of chat_response again, when the earlier logic already allows dummy stream chat responses, making this seem redundant or conflicting?

Reply:
mypy was complaining that it wasn't the correct type

I see the issue though since it could technically return AgentChatResponse

Will need a pr to clean up that logic